common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -103,14 +103,26 @@ public void fetchBlocks(
     try {
       RetryingBlockFetcher.BlockFetchStarter blockFetchStarter =
           (blockIds1, listener1) -> {
+
             // Unless this client is closed.
             if (clientFactory != null) {
-              TransportClient client = clientFactory.createClient(host, port);
+              TransportClient client = null;
+              try {
+                client = clientFactory.createClient(host, port);
+              } catch (Exception e) {
```

Comment:
Are you sure that any exception implies the lost external shuffle service?

Reply:
> Are you sure that any exception implies the lost external shuffle service?

No, that's why I say that we may need a way to check External Shuffle Service alive.
Any good suggestions

Reply:
I don't have any good idea yet.

And if we disallow retry for any exception, why shouldn't we just set to not retry at the beginning?

Reply:
> I don't have any good idea yet.
> 
> And if we disallow retry for any exception, why shouldn't we just set to not retry at the beginning?

Connect failed because of network should retry, but if because of server down, shouldn't retry, I need to find way to check external shuffle service is down.

Reply:
> I don't have any good idea yet.
> 
> And if we disallow retry for any exception, why shouldn't we just set to not retry at the beginning?

How about current check , check host reachable and port can be connected.